app/models/equivalent/uk_callback.rb

```diff
@@ -1,15 +1,17 @@
 module Equivalent
   class UkCallback < Callback
-    def self.options
-      quotas = ApiClient.get_callback_booking_quotas
-      grouped_quotas = quotas.group_by(&:day)
-      options_hash = Hash.new { |hash, key| hash[key] = [] }
-      grouped_quotas.each do |day, data|
-        data.each do |x|
-          options_hash[day] << [x.time_slot, x.start_at]
+    class << self
+      def options
```

Comment:
Why is this different to the method in the parent class? We seem to convert to GMT in that one but not here, but I don't get why? Also this could be DRYer if we just abstract the method that creates the option array and override that instead of the full method

Reply:
I think it is different as for uk callbacks we show BST ?

Reply:
Oh ok, so the x.time_slot is in BST already? makes sense - in that case I would just abstract constructing the option to a new method in the superclass and override that here so you don't need to re-write the whole method

Reply:
have abstracted the grouped_quotas, and set a test. Am happy with the remove_current_day, not sure of the values of putting a function within a loop just to skip it ? Also it is very readable.

Reply:
Can the grouping of options for presentation end up in a helper for use in the view - surely the data model doesn't/shouldn't care how this is getting shown in a select? Just which slot is chosen